kmir/src/kmir/kdist/mir-semantics/ty.k

```diff
@@ -46,283 +51,332 @@ syntax MIRString ::= mirString(String) [group(mir-string), symbol(MIRString::Str
+                   | rigidTyArray(Ty, TyConst)                                    [group(mir-enum), symbol(RigidTy::Array)]
+                   | rigidTyPat(Ty, Pattern)                                      [group(mir-enum), symbol(RigidTy::Pat)]
+                   | rigidTySlice(Ty)                                             [group(mir-enum), symbol(RigidTy::Slice)]
+                   | rigidTyRawPtr(Ty, Mutability)                                [group(mir-enum), symbol(RigidTy::RawPtr)]
+                   | rigidTyRef(Region, Ty, Mutability)                           [group(mir-enum), symbol(RigidTy::Ref)]
+                   | rigidTyFnDef(FnDef, GenericArgs)                             [group(mir-enum), symbol(RigidTy::FnDef)]
+                   | rigidTyFnPtr(PolyFnSig)                                      [group(mir-enum), symbol(RigidTy::FnPtr)]
+                   | rigidTyClosure(ClosureDef, GenericArgs)                      [group(mir-enum), symbol(RigidTy::Closure)]
+                   | rigidTyCoroutine(CoroutineDef, GenericArgs, Movability)      [group(mir-enum), symbol(RigidTy::Coroutine)]
+                   | rigidTyDynamic(ExistentialPredicateBinders, Region, DynKind) [group(mir-enum), symbol(RigidTy::Dynamic)]
+                   | "rigidTyNever"                                               [group(mir-enum), symbol(RigidTy::Never)]
+                   | rigidTyTuple(Tys)                                            [group(mir-enum), symbol(RigidTy::Tuple)]
+                   | rigidTyCoroutineWitness(CoroutineWitnessDef, GenericArgs)    [group(mir-enum), symbol(RigidTy::CoroutineWitness)]
+                   | "rigidTyUnimplemented"                                       [group(mir-enum), symbol(RigidTy::Unimplemented), deprecated] // TODO: remove
+
+syntax TyKind ::= tyKindRigidTy(RigidTy)          [group(mir-enum), symbol(TyKind::RigidTy)]
+                | tyKindAlias(AliasKind, AliasTy) [group(mir-enum), symbol(TyKind::Alias)]
+                | tyKindParam(ParamTy)            [group(mir-enum), symbol(TyKind::Param)]
+                | tyKindBound(MIRInt, BoundTy)    [group(mir-enum), symbol(TyKind::Bound)]
+
+syntax TypeAndMut ::= typeAndMut(ty: Ty, mutability: Mutability) [group(mir---ty--mutability)]
+
+syntax ParamTy ::= paramTy(index: MIRInt, name: MIRString) [group(mir---index--name)]
+syntax BoundTy ::= boundTy(var: MIRInt, kind: BoundTyKind) [group(mir---var--kind)]
+
+// syntax Promoted ::= promoted(Int) [group(mir-int)]
+syntax MaybePromoted ::= promoted(Int) [group(mir-option-int)]
+                       | "notPromoted" [group(mir-option)]
+
+syntax Align ::= align(Int) [group(mir-int), symbol(align)]
+
+// FIXME provSize and allocId are in a _list_ in json
+syntax ProvenanceMapEntry ::= // provenanceMapEntry(provSize: MIRInt, allocId: AllocId) [group(mir), symbol(provenanceMapEntry)]
+                              List [group(mir-klist-MIRInt)]
```

Comment:
I had to change this to a simple list to make it parse. Unsure whether this data structure is used much during execution...

Reply:
Maybe not yet, but it contains information that should be needed. The size is a bad name (named as in rustc internals again(, but it is really an offset, into the allocation with that allocId.